src/python/gudhi/tensorflow/rips_layer.py

```diff
@@ -0,0 +1,75 @@
+import numpy               as np
+import tensorflow          as tf
+from ..rips_complex     import RipsComplex
+
+############################
+# Vietoris-Rips filtration #
+############################
+
+# The parameters of the model are the point coordinates.
+
+def _Rips(DX, max_edge, dimension):
+    # Parameters: DX (distance matrix), 
+    #             max_edge (maximum edge length for Rips filtration), 
+    #             dimension (homology dimension)
+
+    # Compute the persistence pairs with Gudhi
+    rc = RipsComplex(distance_matrix=DX, max_edge_length=max_edge)
+    st = rc.create_simplex_tree(max_dimension=dimension+1)
+    dgm = st.persistence()
+    pairs = st.persistence_pairs()
+
+    # Retrieve vertices v_a and v_b by picking the ones achieving the maximal
+    # distance among all pairwise distances between the simplex vertices
```

Comment:
SimplexTree.flag_persistence_generators already does something very similar to this function, and uses edge_with_same_filtration to find the edge. What is the main goal in reimplementing it (and lower_star_persistence_generators)? Do the existing functions have an inconvenient interface? Do they do unnecessary work that makes them slower? The whole point of those 2 functions was that we could call them for pytorch/tensorflow differentiation. If they don't fit this purpose, maybe we should modify them?

Reply:
My bad, I just forgot that function. I am using it now.

Reply:
There is a similar function for lower-star filtration that could also be nice to use.
Again, if those functions don't quite fit, or if they perform worse than your code, don't hesitate to say something.

Reply:
Thanks! I don't think my code is more optimized than this function, so let's keep consistent and use it as you suggest.